sqlx-core/src/postgres/arguments.rs

```diff
@@ -161,6 +162,24 @@ impl PgArgumentBuffer {
         self.extend_from_slice(&0_u32.to_be_bytes());
         self.type_holes.push((offset, type_name.clone()));
     }
+
+    pub(crate) fn extend_display<T: fmt::Display>(&mut self, value: T) {
+        struct VecFmt<'a>(&'a mut Vec<u8>);
+
+        impl<'a> fmt::Write for VecFmt<'a> {
+            fn write_char(&mut self, c: char) -> fmt::Result {
+                self.write_str(c.encode_utf8(&mut [0u8; 4]))
+            }
+
+            fn write_str(&mut self, s: &str) -> fmt::Result {
+                self.0.extend_from_slice(s.as_bytes());
+                Ok(())
+            }
+        }
+
+        write!(VecFmt(&mut self.buffer), "{}", value)
+            .expect("Display implementation panicked while writing to PgArgumentBuffer");
+    }
```

Comment:
None of this should be necessary. If you import `std::io::Write` instead of `std::fmt::Write` you can use `&mut Vec<u8>` directly as the target of `write!()`.

Kinda confusing, I know. See the examples here: https://doc.rust-lang.org/stable/std/macro.write.html#examples